core/src/main/scala/com/databricks/labs/remorph/preprocessors/jinja/TemplateManager.scala

```diff
@@ -0,0 +1,51 @@
+package com.databricks.labs.remorph.preprocessors.jinja
+
+import com.databricks.labs.remorph.utils.Sed
+
+import scala.collection.mutable
+
+class TemplateManager {
+  private val templates = mutable.Map[String, TemplateElement]()
+  private var counter = 0
```

Comment:
A minor nit: I tend to prefer `private[this]` for fields: by inspection it means I don't have to worry about looking for possible cross-instance access.

This is covered in the Style Guide, along with another reason:
 - https://github.com/databricks/scala-style-guide?tab=readme-ov-file#private-fields
 - https://github.com/databricks/scala-style-guide?tab=readme-ov-file#privatethis